java/vector/src/main/java/org/apache/arrow/vector/ipc/ArrowReader.java

```diff
@@ -216,9 +171,32 @@ private void initialize() throws IOException {
     this.root = new VectorSchemaRoot(schema, vectors, 0);
     this.loader = new VectorLoader(root);
     this.dictionaries = Collections.unmodifiableMap(dictionaries);
+
+    // Read and load all dictionaries from schema
+    for (int i = 0; i < dictionaries.size(); i++) {
```

Comment:
sometimes it's useful to be able to just read the schema out of a message, without loading up any dictionaries or record batches. is there a way to preserve that functionality somehow?

Reply:
Yeah, we could still do that.  I think it just comes down to either reading the dictionaries after the schema, or reading them before the first data batch.  I thought it made a little more sense to read them with the schema, otherwise the user could create the reader, load the schema and try to decode it but fail.

Would it work for you to maybe overload `ArrowReader.readSchema` which will be able to return the original schema before loading the dictionaries?  Similarly, if using the stream format, you could make a subclass of `MessageReader` (introduced here) and react after reading a schema message. If not, I'm ok with reading them before data batches and documenting for the user that you can't decode until batches are read.

Reply:
yeah, an overloaded method would be fine. I agree that having to load a batch before reading dictionaries is a bit confusing for the general use case.